tools/PinVersion/Program.cs

```diff
@@ -12,62 +13,138 @@ namespace PinVersion
+                            if (latestPackage != null)
+                            {
+                                ((JValue)projectJson["version"]).Value = latestPackage.Version.ToNormalizedString();
+                            }
+
+                            var frameworkDependencies = projectJson["frameworks"]
+                                ?.Cast<JProperty>()
+                                ?.Select(f => ((JObject)f.Value)["dependencies"])
+                                ?? Enumerable.Empty<JToken>();
+                            var dependencies = Enumerable
+                                .Concat(new[] { projectJson["dependencies"] }, frameworkDependencies)
+                                .Where(d => d != null)
+                                .SelectMany(d => d)
+                                .Cast<JProperty>();
+
+                            foreach (var dependency in dependencies)
+                            {
+                                latestPackage = packageRepository.FindPackage(dependency.Name);
+                                if (latestPackage != null)
+                                {
+                                    if (dependency.Value.Type == JTokenType.Object)
+                                    {
+                                        // "key": { "version": "1.0.0-*", "type": "build" }
+                                        var value = (JObject)dependency.Value;
+                                        value["version"] = latestPackage.Version.ToNormalizedString();
+                                    }
+                                    else
+                                    {
+                                        // "key": "version"
+                                        dependency.Value = latestPackage.Version.ToNormalizedString();
+                                    }
+                                }
+                            }
+
+                            using (var fileWriter = new JsonTextWriter(new StreamWriter(file)))
+                            {
+                                fileWriter.Formatting = Formatting.Indented;
+                                fileWriter.Indentation = 4;
+                                projectJson.WriteTo(fileWriter);
+                            }
+                        }
+
+                        return 0;
+                    });
+                });
+
+                app.Command("buildscripts", command =>
                 {
-                    latestPackage = packageRepository.FindPackage(dependency.Name);
-                    if (latestPackage != null)
+                    command.Description = "Pins build scripts";
+                    var repositoryPath = command.Argument("[Repository Path]", "Path to the cloned repository");
+                    var koreBuildVersion = command.Argument("[Korebuild Version]", "Korebuild package version");
```

Comment:
1. Could we infer the version for these packages (from the Coherence drop share) rather than specifying them?

2. Does it make sense to have these in the makefile.shade for Universe (which calls this method) rather than change this? The primary reason the pining above was written as an executable and not a sake script was because it needed to reference NuGet.Core \ Json.Net. This piece of code could possibly be written in sake.